keel-veto/src/main/kotlin/com/netflix/spinnaker/keel/veto/unhappy/UnhappyVeto.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.veto.unhappy
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceId
+import com.netflix.spinnaker.keel.api.id
+import com.netflix.spinnaker.keel.persistence.DiffFingerprintRepository
+import com.netflix.spinnaker.keel.persistence.ResourceRepository
+import com.netflix.spinnaker.keel.persistence.ResourceStatus.UNHAPPY
+import com.netflix.spinnaker.keel.persistence.UnhappyVetoRepository
+import com.netflix.spinnaker.keel.veto.Veto
+import com.netflix.spinnaker.keel.veto.VetoResponse
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+
+/**
+ * A veto that stops keel from checking a resource for a configurable
+ * amount of time so that we don't flap on a resource forever.
+ */
+@Component
+class UnhappyVeto(
+  private val resourceRepository: ResourceRepository,
+  private val diffFingerprintRepository: DiffFingerprintRepository,
+  private val unhappyVetoRepository: UnhappyVetoRepository
+) : Veto {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  @Value("veto.unhappy.waiting-time")
+  private var waitingTime: String = "PT10M"
+
+  private val MAX_DIFF_COUNT = 10
```

Comment:
10 seems like a lot for things like large prod server groups. I can imagine situations wherein we'd appreciate if this and `waitingTime` were dynamic properties.

Reply:
Good point. I'll make them optionally dynamic.

Reply:
Ha also I meant for this to be 5.